pkg/v2/crud/crud.go

```diff
@@ -20,7 +21,16 @@ func Add(resource *prop.Resource, path string, value interface{}) error {
 		return err
 	}
 
-	return defaultTraverse(resource.RootProperty(), skipMainSchemaNamespace(resource, head), func(nav prop.Navigator) error {
+	isFound := false
+	err = defaultTraverse(resource.RootProperty(), skipMainSchemaNamespace(resource, head), func(nav prop.Navigator) error {
+		isFound = true
+		return nav.Add(value).Error()
+	})
+	if err != nil || isFound {
+		return err
```

Comment:
do we need to create an error if `isFound`?

Reply:
If the resource is found - it's value is being updated by `nav.Add(value)` -> [Add()](https://github.com/zelloptt/go-scim/blob/5e1a93188f5078d858413865adc87107dc44120c/pkg/v2/prop/navigator.go#L159) -> [stringProperty::Add()](https://github.com/zelloptt/go-scim/blob/5e1a93188f5078d858413865adc87107dc44120c/pkg/v2/prop/string.go#L107)

I haven't check if it's according to the SCIM spec, just left the initial logic untouched.

This PR handles the case when there is no property by the given filter; and therefore the callback function is never called by `defaultTraverse`